File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task134_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
code here:
#include <iostream>
#include <string>
#include <unordered_map>
#include <openssl/sha.h>

class KeyManager {
private:
    std::unordered_map<std::string, std::string> keys;

public:
    std::string generateKey(const std::string& username) {
        unsigned char hash[SHA256_DIGEST_LENGTH];
        SHA256_CTX sha256;
        SHA256_Init(&sha256);
        SHA256_Update(&sha256, username.c_str(), username.size());
        SHA256_Final(hash, &sha256);
        std::string key;
        for (int i = 0; i < SHA256_DIGEST_LENGTH; i++) {
            char buffer[3];
            sprintf(buffer, "%02x", hash[i]);
            key += buffer;
        }
        this->keys[username] = key;
        return key;
    }

    std::string getKey(const std::string& username) {
        return this->keys[username];
    }

    void deleteKey(const std::string& username) {
        this->keys.erase(username);
    }
};

int main() {
    KeyManager manager;
    std::cout << manager.generateKey("user1") << std::endl;
    std::cout << manager.getKey("user1") << std::endl;
    manager.deleteKey("user1");
    std::cout << manager.getKey("user1") << std::endl;
    return 0;
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task134_PERPLEXITY_llama-3-sonar-large-32k-chat.java
code here:
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.HashMap;
import java.util.Map;

public class KeyManager {
    private Map<String, String> keys;

    public KeyManager() {
        this.keys = new HashMap<>();
    }

    public String generateKey(String username) {
        try {
            MessageDigest md = MessageDigest.getInstance("SHA-256");
            byte[] hash = md.digest(username.getBytes());
            StringBuilder hexString = new StringBuilder();
            for (byte b : hash) {
                String hex = Integer.toHexString(0xff & b);
                if (hex.length() == 1) hexString.append('0');
                hexString.append(hex);
            }
            String key = hexString.toString();
            this.keys.put(username, key);
            return key;
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }

    public String getKey(String username) {
        return this.keys.get(username);
    }

    public void deleteKey(String username) {
        this.keys.remove(username);
    }

    public static void main(String[] args) {
        KeyManager manager = new KeyManager();
        System.out.println(manager.generateKey("user1"));
        System.out.println(manager.getKey("user1"));
        manager.deleteKey("user1");
        System.out.println(manager.getKey("user1"));
    }
}